template/lambda-sqs-worker-cdk/infra/appStack.ts

```diff
@@ -51,32 +54,129 @@ export class AppStack extends Stack {
+
     const architecture = '<%- lambdaCdkArchitecture %>';
 
-    const worker = new aws_lambda_nodejs.NodejsFunction(this, 'worker', {
+    const defaultWorkerConfig: aws_lambda_nodejs.NodejsFunctionProps = {
       architecture: aws_lambda.Architecture[architecture],
-      entry: './src/app.ts',
-      bundling: {
-        sourceMap: true,
-        target: 'node20',
-        // By default the aws-sdk-v3 is set as an external module, however, we want it to be bundled with the lambda
-        externalModules: [],
-      },
       runtime: aws_lambda.Runtime.NODEJS_20_X,
-      functionName: '<%- serviceName %>',
       environmentEncryption: kmsKey,
+      // aws-sdk-v3 sets this to true by default so it is not necessary to set the environment variable
+      awsSdkConnectionReuse: false,
+    };
+
+    const defaultWorkerBundlingConfig: aws_lambda_nodejs.BundlingOptions = {
+      sourceMap: true,
+      target: 'node20',
+      // By default the aws-sdk-v3 is set as an external module, however, we want it to be bundled with the lambda
+      externalModules: [],
+    };
+
+    const defaultWorkerEnvironment: Record<string, string> = {
+      NODE_ENV: 'production',
+      // https://nodejs.org/api/cli.html#cli_node_options_options
+      NODE_OPTIONS: '--enable-source-maps',
+    };
+
+    const worker = new aws_lambda_nodejs.NodejsFunction(this, 'worker', {
+      ...defaultWorkerConfig,
+      entry: './src/app.ts',
+      timeout: Duration.seconds(30),
+      bundling: defaultWorkerBundlingConfig,
+      functionName: '<%- serviceName %>',
       environment: {
-        NODE_ENV: 'production',
-        // https://nodejs.org/api/cli.html#cli_node_options_options
-        NODE_OPTIONS: '--enable-source-maps',
+        ...defaultWorkerEnvironment,
         ...context.workerLambda.environment,
       },
-      // aws-sdk-v3 sets this to true by default so it is not necessary to set the environment variable
-      awsSdkConnectionReuse: false,
+      // https://github.com/aws/aws-cdk/issues/28237
+      // This forces the lambda to be updated on every deployment
+      // If you do not wish to use hotswap, you can remove the new Date().toISOString() from the description
+      description: `Lambda description - updated at ${new Date().toISOString()}`,
+    });
+
+    const alias = new aws_lambda.Alias(this, 'worker-live-alias', {
+      aliasName: 'live',
+      version: worker.currentVersion,
+      description: 'The Lambda version currently receiving traffic',
     });
 
-    worker.addEventSource(new aws_lambda_event_sources.SqsEventSource(queue));
+    alias.addEventSource(new aws_lambda_event_sources.SqsEventSource(queue));
 
-    topic.addSubscription(new aws_sns_subscriptions.SqsSubscription(queue));
+    const preHook = new aws_lambda_nodejs.NodejsFunction(
+      this,
+      'worker-pre-hook',
+      {
+        ...defaultWorkerConfig,
+        entry: './src/preHook.ts',
+        timeout: Duration.seconds(30),
```

Comment:
We may want a more generous timeout to allow the generous retry configuration to play out:

https://github.com/seek-oss/skuba/blob/6f0555528126d80a3f035c177697eb1e5de8f03d/template/lambda-sqs-worker-cdk/src/preHook.ts#L11-L19

IIRC this is because CodeDeploy can hang if the hook doesn't provide a well-formed response. Alternatively/additionally we could implement a `Promise.race` or similar to ensure that the hook returns before this timeout.